api/v1alpha1/release_webhook.go

```diff
@@ -17,38 +17,116 @@ limitations under the License.
 
-//+kubebuilder:webhook:path=/validate-appstudio-redhat-com-v1alpha1-release,mutating=false,failurePolicy=fail,sideEffects=None,groups=appstudio.redhat.com,resources=releases,verbs=create;update,versions=v1alpha1,name=vrelease.kb.io,admissionReviewVersions=v1
+// releaseHandler describes the handler for the Release low level webhook
+type releaseHandler struct {
+	Client client.Client
+}
 
-var _ webhook.Validator = &Release{}
+// +kubebuilder:webhook:path=/mutate-appstudio-redhat-com-v1alpha1-release,mutating=true,failurePolicy=fail,sideEffects=None,groups=appstudio.redhat.com,resources=releases,verbs=create;update,versions=v1alpha1,name=mrelease.kb.io,admissionReviewVersions=v1
 
-// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
-func (r *Release) ValidateCreate() error {
-	return nil
+// SetupReleaseLowLevelWebhook registers the Release low level webhook with the passed manager.
+func SetupReleaseLowlevelWebhook(mgr ctrl.Manager) {
+	mgr.GetWebhookServer().Register("/mutate-appstudio-redhat-com-v1alpha1-release", &webhook.Admission{Handler: &releaseHandler{Client: mgr.GetClient()}})
 }
 
-// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
-func (r *Release) ValidateUpdate(old runtime.Object) error {
-	if !reflect.DeepEqual(r.Spec, old.(*Release).Spec) {
-		return fmt.Errorf("release resources spec cannot be updated")
+// Handle takes an incoming admission request and returns an admission response. Create requests add
+// an author label with the current user. Update requests are rejected if the spec or author label
+// are being modified. All other requests are accepted without action.
+func (r *releaseHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	switch req.AdmissionRequest.Operation {
+	case admissionv1.Create:
+		return patchAuthorResponse(req)
+	case admissionv1.Update:
+		var releaseOld, releaseNew Release
+		err := json.Unmarshal(req.Object.Raw, &releaseNew)
+		if err != nil {
+			failedResponse(req.UID, "could not unmarshal the raw incoming Release object")
+		}
+		err = json.Unmarshal(req.OldObject.Raw, &releaseOld)
+		if err != nil {
+			failedResponse(req.UID, "could not unmarshal the raw old Release object")
+		}
+		if !reflect.DeepEqual(releaseNew.Spec, releaseOld.Spec) {
+			return failedResponse(req.UID, "release resources spec cannot be updated")
+		}
+		if releaseNew.GetLabels()[authorLabel] != releaseOld.GetLabels()[authorLabel] {
+			return failedResponse(req.UID, "release author label cannnot be updated")
+		}
 	}
+	return admission.Response{
+		AdmissionResponse: admissionv1.AdmissionResponse{
+			UID:     req.UID,
+			Allowed: true,
+			Result: &metav1.Status{
+				Status: "Success",
+			},
+		},
+	}
+}
 
-	return nil
+// patchAuthorResponse returns a failed admission.Response that blocks the requested operation.
+func failedResponse(uid types.UID, message string) admission.Response {
```

Comment:
This has to be extracted to other place or it will collide with other webhooks redefining it. It can go to operator-goodies maybe... Let's leave it as it is and I'll come up with something.

Reply:
let's revisit this idea when you see new version